src/uPlot.js

```diff
@@ -1761,21 +1816,38 @@ export default function uPlot(opts, data, then) {
 		}
 	}
 
-	function mouseUp(e, src, _x, _y, _w, _h, _i) {
-		if ((e == null || filtMouse(e))) {
+	function mouseUp(e, src, _x, _y, _w, _h, _i, _drag) {
+		const isSyncReq = src != null;
+		let shouldDrag;
+
+		if (isSyncReq || filtMouse(e)) {
 			dragging = false;
 
 			cacheMouse(e, src, _x, _y, _w, _h, _i, false, true);
 
-			if (mouseLeft1 != mouseLeft0 || mouseTop1 != mouseTop0) {
+			let locked = cursor.lock && cursor.locked;
```

Comment:
i think it's sufficient to just check cursor.locked and not need to make a variable?

Reply:
:+1: